src/us/kbase/workspace/kbase/InitWorkspaceServer.java

```diff
@@ -488,56 +484,31 @@ private static AuthToken getHandleToken(
 		return null;
 	}
 	
-
-	private static void checkHandleServiceConnection(
+	private static AbstractHandleClient getHandleServiceClient(
```

Comment:
Have you tested the failure modes for this method manually, since there aren't any tests (and adding tests would be a massive pain?)

Reply:
I didn't test it. yeaaa it would be very painful to test it.

Reply:
You should at least test it manually. What I do is start the tests with a debugger and add a break point after the services are all up but the WS isn't. Then I find the pid of the handle services and kill them and let the tests continue.

Reply:
👍